src/components/ProseMirrorInner.tsx

```diff
@@ -53,7 +53,9 @@ export function ProseMirrorInner({
   });
 
   const editorState =
-    "state" in editorProps ? editorProps.state : editorView?.state ?? null;
+    "defaultState" in editorProps
+      ? editorProps.defaultState
+      : editorProps.state;
```

Comment:
This doesn't look correct to me. This `editorState` variable should be the current editorState from the view (note how before it was being read from `editorView.state` if there was no state prop).

If it's set to the `defaultState`, then it will never update for users that pass a `defaultState`!

Reply:
I had a feeling this was too easy to be true!

Reply:
I _think_ that, instead of this, it would be technically correct to do:

```ts
"defaultState" in editorProps
     // Only use the default state as a fallback for the first render where `editorView` isn't initialized yet
      ? (editorView?.state ?? editorProps.defaultState)
      : editorProps.state;
```

Reply:
so... how does the useState hook react to the state reference that is referenced from `editorView.state`? is it tracked in react state somewhere?

Reply:
This is why there's a forceUpdate call in the useEditorView hook!

Reply:
ah, that makes sense. gonna try your suggestion above!

Reply:
yeah, that seems to work!